MWFTP/clip.py

```diff
@@ -0,0 +1,46 @@
+"""Provides functionality for clipping extreme values.
+
+Based on NormDistCapping.fit_transform().
+"""
+import pandas as pd
+
+
+def clip(data: pd.Series, window_frac: float, stds_clip: float,
+         stds_remove: float) -> pd.DataFrame:
+    """Returns a copy of the series with some of its values possibly removed or
+    clipped, according to their distance from the mean.
+
+    Values stds_remove or more away from the mean are removed. Values stds_clip or
+    more away from mean are set to (mean +- stds_clip)."""
+
+    assert 0 < window_frac <= 1, f'{window_frac}'
+
+    # Find the densest window in the series (lowest span).
+    # We consider only the values within this window for calculating mean and std,
+    # to avoid extreme values.
+    series = data.sort_values()
+    series = series[~series.isna()]
```

Comment:
dropna might be faster as it does not involve indexing

Reply:
Done